samples/navigate-route-with-rerouting/src/main/java/com/esri/arcgismaps/sample/navigateroutewithrerouting/screens/NavigateRouteWithReroutingScreen.kt

```diff
@@ -0,0 +1,181 @@
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.animateContentSize
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.arcgismaps.toolkit.geoviewcompose.rememberLocationDisplay
+import com.esri.arcgismaps.sample.navigateroutewithrerouting.components.NavigateRouteWithReroutingViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.LoadingDialog
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+
+/**
+ * Main screen layout for the sample app
+ */
+@Composable
+fun NavigateRouteWithReroutingScreen(sampleName: String) {
+    val locationDisplay = rememberLocationDisplay()
+    val mapViewModel = viewModel<NavigateRouteWithReroutingViewModel>()
+    // On first composition, initialize the sample.
+    var isViewmodelInitialized by rememberSaveable { mutableStateOf(false) }
+    LaunchedEffect(Unit) {
+        if (!isViewmodelInitialized) {
+            mapViewModel.initialize(locationDisplay)
+            isViewmodelInitialized = true
+        }
+    }
+    // Display loading dialog on initialization
+    if (!isViewmodelInitialized) {
+        LoadingDialog("Loading route result...")
+    }
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) },
+        content = {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it)
+            ) {
+                MapView(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .weight(1f)
+                        .animateContentSize(),
+                    arcGISMap = mapViewModel.arcGISMap,
+                    mapViewProxy = mapViewModel.mapViewProxy,
+                    graphicsOverlays = listOf(mapViewModel.graphicsOverlay),
+                    locationDisplay = locationDisplay
+                )
+                mapViewModel.apply {
```

Comment:
`apply` could be replaced with `with` since you're not configuring an object and returning the object itself.
```suggestion
                with(mapViewModel) {
```